Makefile.config: restore the {OCAMLC,OCAMLOPT}_{CFLAGS,CPPFLAGS} variables

shindere opened the pull request:
They had been renamed in 31cdf416280053dcd38351b7858e818b23110779
(part of #12589) but we should keep the old names for backward compatibility.

This is thus a proposed fix for the issue #13503 and should probably go
through @dra27.

dustanddreams reviewed (approved):
Looking good. Maybe add a future deprecation comment somewhere?

shindere committed:
Makefile.config: restore the {OCAMLC,OCAMLOPT}_{CFLAGS,CPPFLAGS} variables

They had been renamed in 31cdf416280053dcd38351b7858e818b23110779
(part of #12589) but we should keep the old names for backward compatibility.

shindere commented:
Miod Vallat (2024/10/03 09:04 -0700):
> @dustanddreams approved this pull request.

thanks!

> Looking good. Maybe add a future deprecation comment somewhere?

Sure, many thanks for the suggestion.

On top of that I have pushed a commit implementing an alternative
definition of the deprecated variables, defining them in terms of other
build varialbes rather than in terms of tokens to be substituted by
`config.status`.

I am unsure on whihc versions is better. I tend to prefer the first one
because itt's more regular. In the second, alternative one, it is not
possible to define `OCAMLOPT_CPPFLAGS` in temrs of build variables
because we didn't have it before anyway.

dra27 commented on Makefile.config.in:
```diff
@@ -221,6 +221,13 @@ NAKED_POINTERS=false
 
+## Variables deprecated since OCaml 5.3
+
+OCAMLC_CFLAGS=$(BYTECODE_CFLAGS)
+OCAMLOPT_CFLAGS=$(OCAMLOPT_CFLAGS)
+OCAMLC_CPPFLAGS=$(BYTECODE_CPPFLAGS)
+OCAMLOPT_CPPFLAGS=@native_cppflags@
```
It seems OK to have `OCAMLC_CFLAGS` and `OCAMLC_CPPFLAGS` being renamed, and `OCAMLOPT_CFLAGS` and `OCAMLOPT_CPPFLAGS` removed. There's a small typo with it:

```suggestion
OCAMLC_CFLAGS=$(BYTECODE_CFLAGS)
OCAMLOPT_CFLAGS=@native_cflags@
OCAMLC_CPPFLAGS=$(BYTECODE_CPPFLAGS)
OCAMLOPT_CPPFLAGS=@native_cppflags@
```

(I equally don't mind if you just go with the first commit)

shindere commented:
Many thanks for the review, @dra27, and sorry it took me so long to come
back to this one although it's so simple.

> (I equally don't mind if you just go with the first commit)

I went ahead with that, as it was the approach that had (and still has)
my preference.

dra27 reviewed (approved):
Let’s do it!

shindere commented:
David Allsopp (2024/11/06 07:51 -0800):
> @dra27 approved this pull request.
>
> Let’s do it!

Cool, thanks!
